app/policies/group_policy.rb

```diff
@@ -5,26 +5,34 @@ def new?
   alias_method :create?, :new?
 
   def show?
-    user.super_admin? || user.is_organisations_admin?(record.organisation) || user.groups.include?(record)
+    organisation_admin_or_super_admin? || user.groups.include?(record)
   end
 
   def edit?
-    user.super_admin? || user.is_organisations_admin?(record.organisation) || group_admin?
+    organisation_admin_or_super_admin? || group_admin?
   end
 
   alias_method :update?, :edit?
   alias_method :add_editor?, :edit?
 
   def upgrade?
-    user.super_admin? || user.is_organisations_admin?(record.organisation)
+    organisation_admin_or_super_admin?
   end
 
   def request_upgrade?
     group_admin? && !record.active?
   end
 
+  def review_upgrade?
+    organisation_admin_or_super_admin? && record.upgrade_requested?
```

Comment:
For convenience, we are checking that the group is in the correct status for an upgrade request to be reviewed as part of the policy. This allows us to re-use the policy in the show group view to decide which trial group notification banner to display.

This means that if an organisation admin attempts to visit the page to review an upgrade request when the request has already been approved or rejected, they will get a 403 error and see a page with the message "You do not have permission to view this page."

Whilst it might be more ideal to show a more specific error, or a 404 in this scenario this is very much an edge case. For the MVP, we decided not to customise these error pages in the designs but can revisit this later.

Any thoughts on this welcomed.

Reply:
This feels OK for now - I think we've done similar things with policies in the past